src/SpecificDissipationRateEquationSystem.C

```diff
@@ -322,6 +327,7 @@ SpecificDissipationRateEquationSystem::register_inflow_bc(
   SpecDissRate sdr = userData.sdr_;
   std::vector<double> userSpec(1);
   userSpec[0] = sdr.specDissRate_;
+  realm_.sdrFS = userSpec[0];
```

Comment:
I'm still not sure we want this to live on the boundary condition.  Looking at the model it seems like this is a global parameter for the model. Pulling it off the bc user data could be problematic if a user supplies multiple inflow bc's and gives different values.  This will just overwrite the value with the last bc registered.  Since this is really a model constant for the transition model I think we should put it in `SolutionOptions::turbModelConstantMap_`.  I think there should be one entry point for the model 

If the model requires the inflow bc's to all match then we can do a check here to make sure it is within an epsilon of the specified turbulence model constant if the users specifiy it, and if they don't then we can just take the value specified in the `turbModelConstantMap_`.

Reply:
Sorry for the delay in responding to your comment. I had to put this aside due to another project deadline. 

The issue of turbulence inflow conditions is not that straightforward and we may need to have a brief Teams call to fully discuss. However, I will attempt to describe the issue below.

=====================

It is not clear that the present approach of fixing inflow values for k and omega over the region from inlet boundary to the leading edge (LE), can be easily generalized for multiple inflows. Recall, that the objective is to have freestream turbulence quantities of the calculation match those from the experiments at the leading edge of the body. This is difficult to accomplish in the calculation since the freestream turbulence will experience large decay from the inflow to the LE. The approach taken by others and myself here, is to fix the freestream values in a region from inflow to LE.

To explain the problem with multiple inflows, I attached a sketch of an airfoil calculation with two inflows (A & B), which have different inlet values for k & omega. Let's assume that we fixed the bookkeeping problem so that we can input and save conditions for inflows A & B separately. We need turbulence values set at Inflow A to convect to the no slip surface (Region A in the sketch) without being dissipated. Similarly, we need those from Inflow B to do likewise in Region B. The problem is that Regions A & B overlap, so it is not as simple as adding up source terms in Regions A and B.  The turbulence values are not additive in the overlap region.

![IMG_0615](https://user-images.githubusercontent.com/49559493/155890918-d7a20711-a9a7-44ab-a6e8-725c4c52279f.jpg)
.

A more general approach to fix the longstanding problem of decay of the freestream turbulence for transition has recently been proposed in the last several months (Lopes, 2021, attached below). With this approach, the constants of the k-omega dissipation terms (beta and beta_star) are modified outside the boundary layer such that the freestream turbulence doesn't experience decay and target turbulence values at the LE can be achieved more naturally. Inside the boundary layer, those coefficients are not modified so that the original K-omega SST model remains intact.

[Lopes_Decay.pdf](https://github.com/Exawind/nalu-wind/files/8149087/Lopes_Decay.pdf)

I would recommend that the issue of simulations with multiple inflow boundaries be postponed until the approach of Lopes, 2021 or similar can be implemented. I wouldn't be able implement this approach now with my present 4 day a week schedule and other project demands.

Let me know if you have questions or feedback.

Reply:
Here is a link to the full PDF version of the Lopes PhD thesis:

https://www.researchgate.net/publication/353340282_Simulation_of_Transition_from_Laminar_to_Turbulent_Regime_in_Practical_Applications_of_Incompressible_Flow

Reference:
R. Lopes, Simulation of Transition from Laminar to Turbulent Regime in Practical Ap- plications of Incompressible Flow. PhD thesis, Universidade de Lisboa-Instituto Superior T ́ecnico, 2021.

Reply:
@psakievich I have added a check for multiple inflow BCs and pushed my changes. If you approve, could you merge and close this PR?